testapp/build.gradle

```diff
@@ -1,14 +1,14 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.2"
+    compileSdkVersion 21
+    buildToolsVersion "21.1.2"
     useLibrary 'org.apache.http.legacy'
 
     defaultConfig {
         applicationId "com.microsoft.aad.testapp"
         minSdkVersion 18
-        targetSdkVersion 23
+        targetSdkVersion 21
```

Comment:
You're always ready for 23. It's like saying are you clients ready for new version of adal where you were fixing bugs and adding features?